tests/test_widget.py

```diff
@@ -317,3 +316,76 @@ def check_dialog_and_click_no(dialog):
         dialog_action=check_dialog_and_click_no,
         time_out=5,
     )
+
+
+def test_radius_save_preserves_old_data(cpt_widget):
+    """Test saving a new particle radius does not mess up the previous one."""
+    # setup measurement layer
+    # layer_measurements = cpt_widget._setup_measurement_layer()
+    layer_measurements = cpt_widget.viewer.add_points(name="Radii and Lengths", ndim=2)
+
+    # add a new particle and calculate radius
+    cpt_widget.particle_decays_menu.setCurrentIndex(1)
+    # layer_measurements.add([[0, 0, 0, 1], [0, 0, 1, 0], [0, 0, 0, -1]]) # 1px radius
+    layer_measurements.add([[0, 1], [1, 0], [0, -1]])  # 1px radius
+    layer_measurements.selected_data = {0, 1, 2}
+    cpt_widget._on_click_radius()
+
+    # create a second particle and calculate radius
+    cpt_widget.particle_decays_menu.setCurrentIndex(1)
+    # layer_measurements.add([[0, 0, -6, 3], [0, 0, -3, 2], [0, 0, 0, 3]]) # 5px radius
+    layer_measurements.add([[-6, 3], [-3, 2], [0, 3]])  # 5px radius
+    layer_measurements.selected_data = {3, 4, 5}
+    cpt_widget._on_click_radius()
+
+    # check the radius information is different
+    assert (
+        cpt_widget.data[0].radius_px != cpt_widget.data[1].radius_px
+    ), "The radii should be different"
+    assert (
+        cpt_widget.data[0].radius_cm != cpt_widget.data[1].radius_cm
+    ), "The radii should be different"
+    assert (
+        cpt_widget.data[0]._r1 != cpt_widget.data[1]._r1
+    ), "The radius points should be different"
+    assert (
+        cpt_widget.data[0]._r2 != cpt_widget.data[1]._r2
+    ), "The radius points should be different"
+    assert (
+        cpt_widget.data[0]._r3 != cpt_widget.data[1]._r3
+    ), "The radius points should be different"
+
+
+def test_length_save_preserves_old_data(cpt_widget):
+    """Test saving a new particle length does not mess up the previous one."""
+    # setup measurement layer
+    # layer_measurements = cpt_widget._setup_measurement_layer()
+    layer_measurements = cpt_widget.viewer.add_points(name="Radii and Lengths", ndim=2)
+
+    # add a new particle and calculate length
+    cpt_widget.particle_decays_menu.setCurrentIndex(1)
+    # layer_measurements.add([[0, 0, 0, 0], [0, 0, 1, 0]]) # 1px length
+    layer_measurements.add([[0, 0], [1, 0]])  # 1px length
+    layer_measurements.selected_data = {0, 1}
+    cpt_widget._on_click_length()
+
+    # create a second particle and calculate length
+    cpt_widget.particle_decays_menu.setCurrentIndex(1)
+    # layer_measurements.add([[0, 0, 1, 1], [0, 0, 1, 3], [0, 0, 0, 3]]) # 2px length
+    layer_measurements.add([[1, 1], [1, 3]])  # 2px length
+    layer_measurements.selected_data = {2, 3}
+    cpt_widget._on_click_length()
+
+    # check the length information is different
```

Comment:
Ditto my comments about what to comment and what not to.

Comments if the line is opaque and they help explain the test logic. No comments if you've got good variable/function names and you write comments that are just repeating names of the functions.